orchagent/portsorch.cpp

```diff
@@ -1177,6 +1180,64 @@ bool PortsOrch::initPort(const string &alias, const set<int> &lane_set)
     return true;
 }
 
+bool PortsOrch::bake()
+{
+    SWSS_LOG_ENTER();
+
+    // Check the APP_DB port table for warm reboot
+    vector<FieldValueTuple> tuples;
+    bool foundPortConfigDone = m_portTable->get("PortConfigDone", tuples);
+    SWSS_LOG_NOTICE("foundPortConfigDone = %d", foundPortConfigDone);
+
+    bool foundPortInitDone = m_portTable->get("PortInitDone", tuples);
+    SWSS_LOG_NOTICE("foundPortInitDone = %d", foundPortInitDone);
+
+    vector<string> keys;
+    m_portTable->getKeys(keys);
+    SWSS_LOG_NOTICE("m_portTable->getKeys %zd", keys.size());
+
+    if (!foundPortConfigDone || !foundPortInitDone)
+    {
+        SWSS_LOG_NOTICE("No port table, fallback to cold start");
+        cleanPortTable(keys);
```

Comment:
There is #547 to have common function for all swss processes to check whether a process is doing warm start . 
We agreed to discuss the schema design.  The basic idea there is to assume that for cold start, the corresponding WARM_RESTART_TABLE entry cleared, then we create the entry and set restart_count to 0.
Every time the process is warm restarted, restart_count is incremented by 1.   

Not sure about the purpose of cleanPortTable() here.

Reply:
We could still discuss that design. No confliction here.

>> Not sure about the purpose of cleanPortTable() here.
If the redis leftover data for warm reboot is corrupted, clean up everything left and fallback to cold start.

---
In reply to: [205869453](https://github.com/Azure/sonic-swss/pull/551#discussion_r205869453) [](ancestors = 205869453)